service/frontend/workflowHandler.go

```diff
@@ -3312,7 +3313,14 @@ func (wh *WorkflowHandler) getHistory(
 		NextPageToken: nextPageToken,
 		ShardID:       convert.Int32Ptr(shardID),
 	})
-	if err != nil {
+	switch err.(type) {
+	case nil:
+		// noop
+	case *serviceerror.DataLoss:
+		// log event
+		wh.GetLogger().Error("encounter data loss event", tag.WorkflowNamespaceID(namespaceID), tag.WorkflowID(execution.GetWorkflowId()), tag.WorkflowRunID(execution.GetRunId()))
+		return nil, nil, err
+	default:
```

Comment:
So the canonical way to do this is:
```suggestion
	var dataLossErr *serviceerror.DataLoss
	if errors.As(err, &dataLossErr) {
		// log event
		wh.GetLogger().Error("encounter data loss event", tag.WorkflowNamespaceID(namespaceID), tag.WorkflowID(execution.GetWorkflowId()), tag.WorkflowRunID(execution.GetRunId()))
		return nil, nil, err
	}
	if err != nil {
```

Reply:
let us talk, i believe the solution you are suggesting is even verbose